tap_linkedin_ads/sync.py

```diff
@@ -448,379 +93,51 @@ def sync(client, config, catalog, state):
-            for page in sync_analytics_endpoint(client, stream_name, endpoint_config.get('path'), query_string):
-                if page.get(data_key):
-                    responses.append(page.get(data_key))
-        raw_records = merge_responses(responses)
-        time_extracted = utils.now()
-
-        # While we broke the ad_analytics streams out from
-        # `sync_endpoint()`, we want to process them the same. And
-        # transform_json() expects a dictionary with a key equal to
-        # `data_key` and its value is the response from the API
-
-        # Note that `transform_json()` returns the same structure we pass
-        # in. `sync_endpoint()` grabs `data_key` from the return value, so
-        # we mirror that here
-        transformed_data = transform_json({data_key: list(raw_records.values())},
-                                          stream_name)[data_key]
-        if not transformed_data:
-            LOGGER.info('No transformed_data')
-        else:
-            max_bookmark_value, record_count = process_records(
-                catalog=catalog,
-                stream_name=stream_name,
-                records=transformed_data,
-                time_extracted=time_extracted,
-                bookmark_field=bookmark_field,
-                max_bookmark_value=last_datetime,
-                last_datetime=strftime(last_datetime_dt),
-                parent=parent,
-                parent_id=parent_id)
-            LOGGER.info('%s, records processed: %s', stream_name, record_count)
-            LOGGER.info('%s: max_bookmark: %s', stream_name, max_bookmark_value)
-            total_records += record_count
-
-        window_start_date, window_end_date, static_params = shift_sync_window(static_params, today)
-
-        if window_start_date == window_end_date:
-            break
-
-    return total_records, max_bookmark_value
-
-def sync_analytics_endpoint(client, stream_name, path, query_string):
-    page = 1
-    next_url = 'https://api.linkedin.com/v2/{}?{}'.format(path, query_string)
-
-    while next_url:
-        LOGGER.info('URL for %s: %s', stream_name, next_url)
-
-        data = client.get(url=next_url, endpoint=stream_name)
-        yield data
-        next_url = get_next_url(data)
-
-        LOGGER.info('%s: Synced page %s', stream_name, page)
-        page = page + 1
-
-
-def get_next_url(data):
-    next_url = None
-    links = data.get('paging', {}).get('links', [])
-    for link in links:
-        rel = link.get('rel')
-        if rel == 'next':
-            href = link.get('href')
-            if href:
-                next_url = 'https://api.linkedin.com{}'.format(urllib.parse.unquote(href))
-    return next_url
+    # Get the list of parent streams(to sync stream itself or its child stream) for which
+    # sync method needs to be called
+    parent_streams = get_parent_streams(selected_streams)
+
+    # Loop through all parent streams
+    for stream_name in parent_streams:
```

Comment:
What happen if child and parent both are not selected? Instead of parent_streams, we should iterate on sync_to_streams list.

Reply:
In that case, it will not call the sync method for the parent stream also. Here, `parent_streams` contains a stream name if at least parent or child streams are selected. I have updated `parent_streams` name to `sync_to_stream`.